Source/WebKit/WebProcess/WebPage/WebFrame.cpp

```diff
@@ -384,10 +384,15 @@ void WebFrame::transitionToRemote(std::optional<WebCore::LayerHostingContextIden
     auto client = makeUniqueRef<WebRemoteFrameClient>(*this, WTFMove(invalidator));
     auto newFrame = ownerElement
         ? WebCore::RemoteFrame::createSubframeWithContentsInAnotherProcess(*corePage, WTFMove(client), m_frameID, *ownerElement, layerHostingContextIdentifier)
-        : parent ? WebCore::RemoteFrame::createSubframe(*corePage, WTFMove(client), m_frameID, *parent) : WebCore::RemoteFrame::createMainFrame(*corePage, WTFMove(client), m_frameID, localFrame->loader().opener());
+        : parent ? WebCore::RemoteFrame::createSubframe(*corePage, WTFMove(client), m_frameID, *parent) : WebCore::RemoteFrame::createMainFrame(*corePage, WTFMove(client), m_frameID, localFrame->opener());
     if (!parent)
         corePage->setMainFrame(newFrame.copyRef());
     newFrame->takeWindowProxyFrom(*localFrame);
+
+    newFrame->setOpener(localFrame->opener());
```

Comment:
This will also set an opener for subframes, which seems wrong.

Reply:
But only if the frame we're replacing already has an opener, which shouldn't happen.

Reply:
I see. Is this needed at all then? `localFrame->opener()` is already passed to `RemoteFrame::createMainFrame()`.